src/modules/expression/unop/neg.rs

```diff
@@ -51,6 +57,21 @@ impl TranslateModule for Neg {
     }
 }
 
+impl Neg {
+    pub fn get_integer_value(&self) -> Option<isize> {
+        self.expr.get_integer_value().map(isize::neg)
+    }
+
+    pub fn get_array_index(&self, meta: &mut TranslateMetadata) -> String {
+        if let Some(expr) = self.get_integer_value() {
+            expr.to_string()
+        } else {
+            let expr = self.expr.translate(meta);
+            translate_computation(meta, ArithOp::Neg, None, Some(expr))
+        }
+    }
+}
```

Comment:
Shouldn't this function be the default `translate` function? We might want to get the optimized version regardless of the situation, right?

Reply:
Ah, good spot. Have replaced duplicated code with call to default function.

Reply:
Maybe we can revisit this if/when we have a separate integer type.

Reply:
Won't fix, for reasons given.